backend/question-service/controllers/historyController.ts

```diff
@@ -0,0 +1,178 @@
+import { Request, Response } from 'express';
+import jwt, { JwtPayload } from 'jsonwebtoken';
+import historyEntryModel from '../models/HistoryEntry';
+import { Question } from 'models/Question';
+
+const getErrorMessage = (error: unknown): string => {
+  if (error instanceof Error) return error.message;
+  return 'An unexpected error occurred';
+};
+
+const extractUserIdFromToken = (req: Request): string | null => {
```

Comment:
I dont think we need this method.  We can access the id from `req` straightaway since `authenticateToken` already does `req.userId = decoded.id`

Reply:
Sorry, but you're gonna have to explain this better.